client/client_execute.go

```diff
@@ -0,0 +1,146 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// Execute executes http request as described in Req.
+//
+// If Req.WriteTo is specified, it will also populate the resultContainer
+func (r *Request) Execute(ctx context.Context, req *Req) ([]byte, error) {
+	request, err := r.constructHttpRequest(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	startTime := time.Now()
+	res, err := r.HttpClient.Do(request)
+	r.reportMonitoringMetricsIfEnabled(startTime, request, req, res, err)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.HttpErrorHandler(res, request.URL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode >= http.StatusBadRequest {
```

Comment:
I wonder why we need this if we have error handler a few lines above ? Might be something to improve in future.

Reply:
errorHandler is [passed in from constructor](https://github.com/trustwallet/go-libs/pull/177/files#diff-bd3a55a72186f59e2e63efb4951573b2f9e4a7cc98086e922b0859f8ccc1dd09R46)

It is used for logging to sentry AFAIK